sdk/spring/spring-cloud-azure-actuator/src/main/java/com/azure/spring/cloud/actuator/storage/StorageBlobHealthIndicator.java

```diff
@@ -21,35 +17,33 @@
 public class StorageBlobHealthIndicator extends AbstractHealthIndicator {
 
     private final BlobServiceAsyncClient blobServiceAsyncClient;
-    private Duration timeout = DEFAULT_HEALTH_CHECK_TIMEOUT;
+
+    private final ResourceLoader resourceLoader;
 
     /**
      * Creates a new instance of {@link StorageBlobHealthIndicator}.
+     *
      * @param blobServiceAsyncClient the blob service client
+     * @param resourceLoader the resource loader
      */
-    public StorageBlobHealthIndicator(BlobServiceAsyncClient blobServiceAsyncClient) {
+    public StorageBlobHealthIndicator(BlobServiceAsyncClient blobServiceAsyncClient, ResourceLoader resourceLoader) {
         this.blobServiceAsyncClient = blobServiceAsyncClient;
+        this.resourceLoader = resourceLoader;
     }
 
     @Override
-    protected void doHealthCheck(Health.Builder builder) throws Exception {
+    protected void doHealthCheck(Health.Builder builder) {
         if (blobServiceAsyncClient == null) {
             builder.status(NOT_CONFIGURED_STATUS);
         } else {
             builder.withDetail(URL_FIELD, blobServiceAsyncClient.getAccountUrl());
-            final Response<BlobServiceProperties> info = blobServiceAsyncClient.getPropertiesWithResponse()
-                                                                               .block(timeout);
-            if (info != null) {
+            try {
+                resourceLoader.getResource("azure-blob://spring-cloud-azure-not-existing-container/not-existing-blob");
                 builder.up();
+            } catch (Exception e) {
+                builder.down();
             }
         }
     }
 
-    /**
-     * Set health check request timeout.
-     * @param timeout the duration value.
-     */
-    public void setTimeout(Duration timeout) {
```

Comment:
To avoid breaking change, mark it as `deprecated` instead of deleting it.

Reply:
If we use the SDK API directly, we probably still need this method?

Reply:
Agree.

Reply:
I will revert this change. We donot have an API to do this in data plane in this case. We will decide to accept it or work around it.